node/index.d.ts

```diff
@@ -1891,8 +1891,27 @@ declare module "dns" {
         priority: number
     }
 
-    export function lookup(domain: string, family: number, callback: (err: Error, address: string, family: number) => void): string;
-    export function lookup(domain: string, callback: (err: Error, address: string, family: number) => void): string;
+    // Supported getaddrinfo flags
+    export const ADDRCONFIG: number;
+    export const V4MAPPED: number;
+
+    export interface LookupOptions {
+        family?: number;
+        hints?: number;
+        all?: boolean;
+    }
+
+    interface Address {
+        address: string;
+        family: number;
+    }
+
+    interface QueryReqWrap {}
+
+    export function lookup(domain: string, family: number, callback: (err: NodeJS.ErrnoException, address: string, family: number) => void): QueryReqWrap;
+    export function lookup(domain: string, options: LookupOptions, callback: (err: NodeJS.ErrnoException, address: string, family: number) => void): QueryReqWrap;
+    export function lookup(domain: string, options: LookupOptions, callback: (err: NodeJS.ErrnoException, addresses: Address[]) => void): QueryReqWrap;
```

Comment:
I revisited current declarations. There are 3 cases:
1. Option `all` doesn't exist or `false`, then callback is `(err: NodeJS.ErrnoException, address: string, family: number) => void`.
2. Option `all` is `true`, then callback is `(err: NodeJS.ErrnoException, addresses: Address[]) => void`.
3. Options `all` has `boolean` type (for example user assign value of some function where we don't known exact literal type (`true` or `false`) at compile time), then callback is union of 1 and 2.

In typescript:

```ts
interface LookupOptions {
    family?: number;
    hints?: number;
    all?: boolean;
}

interface LookupOneOptions extends LookupOptions {
    family?: number;
    hints?: number;
    all?: false;
}

interface LookupAllOptions extends LookupOptions {
    family?: number;
    hints?: number;
    all: true;
}

export function lookup(domain: string, options: LookupOneOptions, callback: (err: NodeJS.ErrnoException, address: string, family: number) => void): void;
export function lookup(domain: string, options: LookupAllOptions, callback: (err: NodeJS.ErrnoException, addresses: Address[]) => void): void;
export function lookup(domain: string, options: LookupOptions, callback: (err: NodeJS.ErrnoException, address: string | Address[], family?: number) => void): void;
```
What do you think about this?